signature/mechanism_sigstore.go

```diff
@@ -0,0 +1,325 @@
+			Data: &models.RekordV001SchemaData{
+				Content: strfmt.Base64(payload),
+			},
+			Signature: &models.RekordV001SchemaSignature{
+				Content: strfmt.Base64(signature),
+				Format:  models.RekordV001SchemaSignatureFormatX509,
+				PublicKey: &models.RekordV001SchemaSignaturePublicKey{
+					Content: strfmt.Base64(pubKey),
+				},
+			},
+		},
+	}
+}
+
+func verifyTLogEntry(rekorClient *client.Rekor, uuid string) (*models.LogEntryAnon, error) {
+	params := entries.NewGetLogEntryByUUIDParams()
+	params.EntryUUID = uuid
+
+	lep, err := rekorClient.Entries.GetLogEntryByUUID(params)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(lep.Payload) != 1 {
+		return nil, errors.New("UUID value can not be extracted")
+	}
+	e := lep.Payload[params.EntryUUID]
+
+	if e.Verification == nil || e.Verification.InclusionProof == nil {
+		return nil, fmt.Errorf("inclusion proof not provided")
+	}
+
+	hashes := [][]byte{}
+	for _, h := range e.Verification.InclusionProof.Hashes {
+		hb, _ := hex.DecodeString(h)
+		hashes = append(hashes, hb)
+	}
+
+	rootHash, _ := hex.DecodeString(*e.Verification.InclusionProof.RootHash)
+	leafHash, _ := hex.DecodeString(params.EntryUUID)
+
+	v := logverifier.New(hasher.DefaultHasher)
+	if err := v.VerifyInclusionProof(*e.Verification.InclusionProof.LogIndex, *e.Verification.InclusionProof.TreeSize, hashes, rootHash, leafHash); err != nil {
+		return nil, errors.Wrap(err, "verifying inclusion proof")
+	}
+
+	// Verify rekor's signature over the SET.
+	resp, err := rekorClient.Pubkey.GetPublicKey(pubkey.NewGetPublicKeyParams())
+	if err != nil {
+		return nil, errors.Wrap(err, "rekor public key")
+	}
+	rekorPubKey, err := PemToECDSAKey([]byte(resp.Payload))
+	if err != nil {
+		return nil, errors.Wrap(err, "rekor public key pem to ecdsa")
+	}
+
+	payload := BundlePayload{
+		Body:           e.Body,
+		IntegratedTime: *e.IntegratedTime,
+		LogIndex:       *e.LogIndex,
+		LogID:          *e.LogID,
+	}
+	if err := VerifySET(payload, []byte(e.Verification.SignedEntryTimestamp), rekorPubKey); err != nil {
+		return nil, errors.Wrap(err, "verifying signedEntryTimestamp")
+	}
+
+	return &e, nil
+}
+
+func PemToECDSAKey(pemBytes []byte) (*ecdsa.PublicKey, error) {
+	pub, err := cryptoutils.UnmarshalPEMToPublicKey(pemBytes)
+	if err != nil {
+		return nil, err
+	}
+	ecdsaPub, ok := pub.(*ecdsa.PublicKey)
+	if !ok {
+		return nil, fmt.Errorf("invalid public key: was %T, require *ecdsa.PublicKey", pub)
+	}
+	return ecdsaPub, nil
+}
+
+type BundlePayload struct {
+	Body           interface{} `json:"body"`
+	IntegratedTime int64       `json:"integratedTime"`
+	LogIndex       int64       `json:"logIndex"`
+	LogID          string      `json:"logID"`
+}
+
+func VerifySET(bundlePayload BundlePayload, signature []byte, pub *ecdsa.PublicKey) error {
+	contents, err := json.Marshal(bundlePayload)
+	if err != nil {
+		return errors.Wrap(err, "marshaling")
+	}
+	canonicalized, err := jsoncanonicalizer.Transform(contents)
```

Comment:
I'd love avoid new dependencies if possible.  This one seems to be maintained by a single person which is something we should consider.  In case we need to canonicalize, maybe we can use another library; ideally one we already use.